wolfcrypt/src/asn.c

```diff
@@ -24174,14 +24180,23 @@ static int EncodeCert(Cert* cert, DerCert* der, RsaKey* rsaKey, ecc_key* eccKey,
 
     /* CA */
     if (cert->isCA) {
-        der->caSz = SetCa(der->ca, sizeof(der->ca));
+        der->caSz = SetCa(der->ca, sizeof(der->ca), cert->isCA);
         if (der->caSz <= 0)
             return CA_TRUE_E;
 
         der->extensionsSz += der->caSz;
     }
-    else
-        der->caSz = 0;
+    
+    /* if isCA is false but Basic Constraint is set */
+    else if(cert->decodedCert != NULL) {
+        if(((DecodedCert*)cert->decodedCert)->extBasicConstSet){ 
```

Comment:
Please add space `if (` and `) {`. Consistency with existing code and our coding standards.